lib/pengine/utils.c

```diff
@@ -834,10 +834,17 @@ unpack_operation(action_t * action, xmlNode * xml_obj, resource_t * container,
         } else {
             value = "recover baremetal remote node connection (default)";
         }
-        if (action->rsc->remote_reconnect_interval) {
+
+        if ((action->rsc->remote_reconnect_interval) ||
+            !is_set(action->rsc->flags, pe_rsc_managed)) {
             action->fail_role = RSC_ROLE_STOPPED;
         }
-        action->on_fail = action_fail_reset_remote;
+
+        if (is_set(action->rsc->flags, pe_rsc_managed)) {
+            action->on_fail = action_fail_reset_remote;
+        } else {
+            action->on_fail = action_fail_stop;
+        }
```

Comment:
This section provides a default on-fail if the user doesn't specify it. What should we do if the user does specify on-fail for an operation that fails when the resource is unmanaged? Should we override the user's value in that case?